lib/collection/src/operations/types.rs

```diff
@@ -1345,10 +1345,27 @@ impl From<tonic::Status> for CollectionError {
             tonic::Code::FailedPrecondition => CollectionError::PreConditionFailed {
                 description: format!("{err}"),
             },
-            tonic::Code::ResourceExhausted => CollectionError::RateLimitExceeded {
-                description: format!("{err}"),
-                retry_after: None,
-            },
+            tonic::Code::ResourceExhausted => {
+                // extract retry-after from metadata
+                // the value is passed as a String containing an integer number of seconds
+                let retry_after = err.metadata().get("retry-after").and_then(|v| {
+                    v.to_str()
+                        .inspect_err(|e| log::info!("Failed to parse retry-after header: {e}"))
+                        .ok()
+                        .and_then(|v| {
+                            v.parse::<u64>()
+                                .inspect_err(|e| {
+                                    log::info!("Failed to parse retry-after header: {e}")
```

Comment:
Nit:

```suggestion
                                    log::info!("Failed to parse retry-after value in gRPC metadata (value: {v}): {e}")
```

May point to the problem a bit better. Also, the error does not include the malformed value, so it might be useful to include it explicitly.